src/services/orders.service.ts

```diff
@@ -0,0 +1,74 @@
+import Stripe from 'stripe';
+import XLSX from 'xlsx';
+import { FastifyBaseLogger, FastifyInstance } from 'fastify';
+import { BadRequestError, CustomError, InternalServerError } from '../custom-errors';
+
+interface OrderCheckResult {
+  orderId: string;
+  type: 'payment_intent' | 'subscription';
+  refunded?: boolean | null;
+}
+
+export async function processOrderId(
+  orderId: string,
+  stripe: Stripe,
+  log: FastifyInstance['log'],
+): Promise<OrderCheckResult> {
+  try {
+    if (!orderId.startsWith('pi_')) {
+      return { orderId, type: 'subscription', refunded: null };
+    }
+
+    const paymentIntent = await stripe.paymentIntents.retrieve(orderId, {
+      expand: ['latest_charge'],
+    });
+
+    const refunded = paymentIntent.status === 'succeeded' && (paymentIntent.latest_charge as Stripe.Charge).refunded;
+
+    return {
+      orderId,
+      type: 'payment_intent',
+      refunded,
+    };
+  } catch (error) {
+    log.error(`Error processing order ID ${orderId}: ${(error as Error).message}`);
+    throw new InternalServerError(`Failed to process order ID ${orderId}`);
+  }
+}
+
+export async function processUploadedFile(
```

Comment:
Add a JSDoc specifying which is the expect file format so anyone can be referred to that explanation without having to guess how this function handles it